packages/cli/src/databases/migrations/common/1732549866705-CreateTestRunTable.ts

```diff
@@ -0,0 +1,27 @@
+import type { MigrationContext, ReversibleMigration } from '@/databases/types';
+
+const testRunTableName = 'test_run';
+
+export class CreateTestRun1732549866705 implements ReversibleMigration {
+	async up({ schemaBuilder: { createTable, column } }: MigrationContext) {
+		await createTable(testRunTableName)
+			.withColumns(
+				column('id').varchar(36).primary.notNull,
+				column('testDefinitionId').varchar(36).notNull,
+				column('status').varchar().notNull,
+				column('runAt').timestamp(),
+				column('completedAt').timestamp(),
+				column('metrics').json,
+			)
+			.withIndexOn('testDefinitionId')
+			.withForeignKey('testDefinitionId', {
+				tableName: 'test_definition',
+				columnName: 'id',
+				onDelete: 'CASCADE',
+			}).withTimestamps;
+	}
```

Comment:
It would be nice to enforce that the correct timestamps have been set in specific statuses. I.e. add a CHECK constraint:

```sql
CHECK (
  (status = 'new' AND runAt IS NULL AND completedAt IS NULL) OR
  (status = 'running' AND runAt IS NOT NULL AND completedAt IS NULL) OR
  (status = 'completed' AND runAt IS NOT NULL AND completedAt IS NOT NULL) OR
  (status = 'error' AND which timestamps in this case?)
)
```

This would make sure we don't have invalid states.

Reply:
Thanks, this is a great suggestion. The list of statuses isn't settled down yet, but once it is - we will add constraints